rust/src/bin/redis_playerdata_save_load.rs

```diff
@@ -31,94 +69,39 @@ fn main() -> BoxResult<()> {
     }
     CombinedLogger::init(multiple).unwrap();
 
-    let mut args: Vec<String> = env::args().collect();
+    let args = Opts::parse();
 
-    if args.len() < 5 {
-        usage();
-        return Ok(());
-    }
-
-    args.remove(0); // Program name
-
-    let redis_uri = args.remove(0);
-
-    let domain = args.remove(0);
-
-    let mode = match &args.remove(0)[..] {
-        "--input" => Mode::INPUT,
-        "--output" => Mode::OUTPUT,
-        _ => {
-            usage();
-            return Ok(());
+    // Make sure directory exists / does not exist as appropriate for command
+    if let Commands::OUTPUT(outargs) = &args.command {
+        if Path::new(&outargs.path).is_dir() {
+            bail!("Supplied output directory already exists!");
         }
-    };
-
-    // Get the path to the input/output directory
-    let basedir = args.remove(0);
-    let basedir = Path::new(&basedir);
-    if mode == Mode::OUTPUT && basedir.is_dir() {
-        bail!("Supplied output directory already exists!");
-    } else if mode == Mode::INPUT && !basedir.is_dir() {
-        bail!("Supplied input directory does not exist");
     }
-
-    // Default to keeping all history elements
-    let mut to_keep : isize = 999999;
-    if Mode::INPUT == mode {
-        if args.len() > 0 {
-            to_keep = args.remove(0).parse::<isize>().unwrap();
-        }
-        if to_keep < 0 {
-            bail!("Number of history elements to keep must be 1 or more");
```

Comment:
Looks like this wasn't properly checked before at any rate, but the message says 1 or more, and the minimum value in both versions is 0.